### PR TITLE
tidy: Use more `cargo-deny` features

### LINKED ISSUE #34393: Migrate duplicate version checks to cargo-deny
We just need to move entries from servo-tidy.toml to deny.toml and remove python code.

### PR DESCRIPTION
**Pull request by mrobinson**
Instead of parsing the `Cargo.lock` file directly in `tidy.py`. Use
`cargo-deny`, which we already use to detect unapproved licenses in the
dependency chain to detect duplicate and banned crates. In addition,
enable all other `cargo-deny` checks and add exceptions where necessary
for them. This depends on the latest release of `cargo-deny` which
depends on a recent verison of `rust`.

Fixes #34393.

Signed-off-by: Martin Robinson <mrobinson@igalia.com>

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #34393.
- [x] These changes do not require tests because the servo tidy integration with `cargo-deny` does not currently have testing support.

### CODE REVIEW

**Comment by mrobinson** [comment]
> > This depends on the latest release of cargo-deny which depends on a recent verison of
> > rust, so it must wait for the rustup.
> 
> Installing binary crates doesn't need to be tied to the rust version we pin for servo. I think the Github hosted runners already include Rust 1.82 by default, so we wouldn't even need to install another rust toolchain. I'm not sure about our self hosted runners though.

This is true, but for those simply following the build instructions, they are only guaranteed to have a single rust version installed. It seems good to standardize on that single rust version for all of our tooling.

**Comment by mrobinson** [comment]
This should now be ready for review.

**Review by jschwe — APPROVED** [review]

File: python/servo/platform/base.py [hunk 1]
```diff
@@ -82,8 +82,9 @@ def install_cargo_deny(self, force: bool) -> bool:
         if not force and shutil.which("cargo-deny") is not None:
```
Comment (jschwe) [inline python/servo/platform/base.py:82]: We should also check the version here if we want to ensure >= 0.16.3 is installed. 
For me ./mach test-tidy did not complain or fail, although I still have 0.14 installed

Comment (mrobinson) [reply python/servo/platform/base.py]: Okay. I've made this check. One still needs to run `./mach bootstrap` again though.

File: Cargo.lock [hunk 2]
```diff
@@ -7768,9 +7768,9 @@ dependencies = [
 
 [[package]]
 name = "url"
-version = "2.5.3"
+version = "2.5.4"
```
Comment (jdm) [inline Cargo.lock:7771]: This is the source of the unexpected test failures.

Comment (mrobinson) [reply Cargo.lock]: Oh, interesting. I've undone the upgrade and added an exception for `url@2.5.3` which has apparently been yanked.

**Pull request merged**